----

# return status
float32[] progress

